src/nvim/fold.c

```diff
@@ -1505,36 +1486,40 @@ static int getDeepestNestingRecurse(garray_T *gap)
 /*
  * Check if a fold is closed and update the info needed to check nested folds.
  */
-static int
-check_closed(
-    win_T *win,
-    fold_T *fp,
-    int *use_levelp,            // TRUE: outer fold had FD_LEVEL
-    int level,                  // folding depth
-    int *maybe_smallp,          // TRUE: outer this had fd_small == MAYBE
-    linenr_T lnum_off           // line number offset for fp->fd_top
+static bool check_closed(
+    win_T *const win,
+    fold_T *const fp,
+    bool *const use_levelp,     // true: outer fold had FD_LEVEL
+    const int level,            // folding depth
+    bool *const maybe_smallp,   // true: outer this had fd_small == kNone
```

Comment:
It looks like you `const`'d the pointer value rather than the pointer contents. Why?

Reply:
Adding `const` to pointer value ensures that it doesn't change in the function and it's safe to pass to functions with non-const pointer params without typecast. C is mutable by default, including the params, so adding `const` gives some assurance that it doesn't change midway in a long function.

Reply:
Adding const to every local binding that happens to not change will add a lot of linenoise to the implementation. Overly long functions (so long that local vars are hard to keep track of) are the exception, they at least shouldn't determine how we write reasonably sized functions. (Didn't check this function because of on mobile)